Canvas.cpp

```diff
@@ -12,15 +14,24 @@ Canvas::Canvas(QWidget *parent) : QWidget(parent)
 Canvas::~Canvas()
 {
     // Free all shapes when deleted. Vector class will delete dynamic array
-    for (auto shapePtr : this->shapes)
-    {
-        delete shapePtr;
-    }
+    clearShapes();
 }
 
 Vector<Shape*> Canvas::getShapes()
 {
-    return this->shapes;
+    return shapes;
+}
+
+void Canvas::drawShapesInFile(QString &fileDir)
```

Comment:
I personally would prefer `loadFromFile` to make the direction more clear, this sounds like it could be saving to a file